src/BenchmarkDotNet/Toolchains/DotNetCli/DotNetCliBuilder.cs

```diff
@@ -32,7 +32,8 @@ public BuildResult Build(GenerateResult generateResult, BuildPartition buildPart
                     buildPartition,
                     Array.Empty<EnvironmentVariable>(),
                     buildPartition.Timeout,
-                    logOutput: LogOutput)
+                    logOutput: LogOutput,
+                    retryFailedBuildWithNoDeps: false)
```

Comment:
`DotNetCliBuilder` is used by other toolchains than WASM. Please set the value to `true` by default (to keep current behavior) and to `false` for the WASM toolchain:

https://github.com/dotnet/BenchmarkDotNet/blob/1d34f856a4ba633af9efc63394abbd0311edfbae/src/BenchmarkDotNet/Toolchains/MonoWasm/WasmToolChain.cs#L43